upload/inc/plugins/pinpost.php

```diff
@@ -205,12 +225,32 @@ function pinpost_populate(&$post)
 	$allowed_forums = explode(',', $mybb->settings['pinpost_forums']);
 
 	if (in_array($post['fid'], $allowed_forums) || in_array('-1', $allowed_forums)) {
-		global $db, $templates, $lang, $thread;
+		global $db, $templates, $lang, $thread, $visible, $pinnedposts;
```

Comment:
Nice improvement, just one query, what is the usage of `$pinnedposts` ?

Reply:
Currently pins are rendered within the `build_postbit()` scope, this will duplicate the result to be used in the `showthread` template.

I didn't add it to the template replacements to avoid duplicating it but I didn't intend to remove your variable either.

Reply:
Yes, The scope.
But isn't the `$post` itself is in showthread scope? So if I amend `$post` the addition is available in showthread as well. No?
Anyway; it doesn't harm a thing, just for my understanding.

Reply:
`$post` isn't available at the `showthread` template.